src/OrchardCore.Modules/OrchardCore.Forms/Shapes.cs

```diff
@@ -0,0 +1,123 @@
+using OrchardCore.DisplayManagement;
+using OrchardCore.DisplayManagement.Descriptors;
+
+namespace OrchardCore.Forms
+{
+    public class Shapes : IShapeTableProvider
+    {
+        public void Discover(ShapeTableBuilder builder)
+        {
+            builder.Describe("ContentCard")
+            .OnCreated(async context =>
+            {
+
+                dynamic cardShape = context.Shape;
+
+                IShape settingTool = await context.ShapeFactory.New.ToolFormSettings();
+                settingTool.Metadata.Name = "ToolFormSettings";
+
+                settingTool.Properties["Content"] = await context.ShapeFactory.CreateAsync("ContentZone");
+                settingTool.Properties["Actions"] = await context.ShapeFactory.CreateAsync("ContentZone");
+
+                cardShape.Footer.Add(settingTool, "before");
+
+
+                IShape formWorkflowTool = await context.ShapeFactory.New.ToolFormWorkflow();
+                formWorkflowTool.Metadata.Name = "ToolFormWorkflow";
+
+                formWorkflowTool.Properties["Content"] = await context.ShapeFactory.CreateAsync("ContentZone");
+                formWorkflowTool.Properties["Actions"] = await context.ShapeFactory.CreateAsync("ContentZone");
+
+
+                cardShape.Footer.Add(formWorkflowTool, "before");
+
+            })
+            .OnDisplaying(context =>
+            {
+                dynamic cardShape = context.Shape;
+
+                bool hasSettingsTool = false;
+
+                if (cardShape.ContentItem.ContentType == "Form")
+                {
+                    //Move WorkflowSettings to ToolFormWorkflow Content, if any
+                    foreach (var item in cardShape.ContentEditor.WorkflowSettings)
+                    {
+                        cardShape.Footer.ToolFormWorkflow.Content.Add(item);
+                    }
+                    cardShape.ContentEditor.WorkflowSettings = null;
+
+                    //Configure ToolFormWorkflow Shape
+                    cardShape.Footer.ToolFormWorkflow.ModalId = $"ToolFormWorkflow_{cardShape.ContentItem.ContentItemId}";
+                    cardShape.Footer.ToolFormWorkflow.ContentItemDisplayText = cardShape.ContentItem.DisplayText;
+                    cardShape.Footer.ToolFormWorkflow.ContentType = cardShape.ContentItem.ContentType;
+
+                    hasSettingsTool = true;
+                }
+                else
+                {
+                    // Remove Workflow Tool
```

Comment:
We can't be adding and removing shapes here like you have done. I remember the issue you opened about it. I think we need to resolve that

Reply:
Yes, currently there is no way to access params in `OnCreated` #6212, So I create shape in OnCreated and remove it based on params.

I also had to explicitly give it a Metadaa.Name as its null #6211